server/svix-server/tests/e2e_attempt.rs

```diff
@@ -145,3 +145,191 @@ async fn test_list_attempts_by_endpoint() {
+
+        let last_three_iter_field: ListResponse<MessageAttemptOut> = client
+            .get(
+                &format!(
+                    "api/v1/app/{}/attempt/endpoint/{}/?limit=3&iterator={}",
+                    app.id,
+                    ep.id,
+                    first_three.iterator.unwrap()
+                ),
+                StatusCode::OK,
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(last_three_manual.data, last_three_iter_field.data);
+
+        assert_eq!(last_three_manual.data.len(), 3);
+        assert_eq!(all_attempts.data[3], last_three_manual.data[0]);
+        assert_eq!(all_attempts.data[4], last_three_manual.data[1]);
+        assert_eq!(all_attempts.data[5], last_three_manual.data[2]);
+        assert!(last_three_manual.done);
+
+        // `prev` iterator
+        let two_and_three: ListResponse<MessageAttemptOut> = client
+            .get(
+                &format!(
+                    "api/v1/app/{}/attempt/endpoint/{}/?limit=2&iterator={}",
+                    app.id,
+                    ep.id,
+                    last_three_manual.prev_iterator.unwrap()
+                ),
+                StatusCode::OK,
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(two_and_three.data.len(), 2);
+        assert_eq!(all_attempts.data[1], two_and_three.data[0]);
+        assert_eq!(all_attempts.data[2], two_and_three.data[1]);
+        assert!(!two_and_three.done);
+
+        let one: ListResponse<MessageAttemptOut> = client
+            .get(
+                &format!(
+                    "api/v1/app/{}/attempt/endpoint/{}/?limit=2&iterator={}",
+                    app.id,
+                    ep.id,
+                    two_and_three.prev_iterator.unwrap()
+                ),
+                StatusCode::OK,
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(one.data.len(), 1);
+        assert_eq!(all_attempts.data[0], one.data[0]);
+        assert!(one.done);
+
+        // `after` field
+        let first_three_by_time: ListResponse<MessageAttemptOut> = client
+            .get(
+                &format!(
+                    "api/v1/app/{}/attempt/endpoint/{}/?after={}",
+                    app.id, ep.id, all_attempts.data[3].created_at,
+                ),
+                StatusCode::OK,
+            )
+            .await
+            .unwrap();
+        assert_eq!(first_three_by_time.data.len(), 3);
+        assert_eq!(all_attempts.data[0], first_three_by_time.data[0]);
+        assert_eq!(all_attempts.data[1], first_three_by_time.data[1]);
+        assert_eq!(all_attempts.data[2], first_three_by_time.data[2]);
+
+        // `before field`
+        let last_three_by_time: ListResponse<MessageAttemptOut> = client
+            .get(
+                &format!(
+                    "api/v1/app/{}/attempt/endpoint/{}/?before={}",
+                    app.id, ep.id, all_attempts.data[2].created_at,
```

Comment:
Does this actually work? I've had issues with doing something like this because `created_at` doesn't get url-escaped (it has a colon in it).

Reply:
It could be that the tests are wrong. :)

For example, it could be that you wrote `before` as `befffore` and it's not actually parsing the correct thing and that's why it works. Need to make sure it's correct by e.g. adding a test that uses a timestamp from before that expects 0 results or something of that sort.